libcudacxx/include/cuda/std/detail/libcxx/include/complex

```diff
@@ -292,6 +292,25 @@ template<class T> complex<T> tanh (const complex<T>&);
 
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
+template<class _Tp>
+struct __is_complex_float {
+    static constexpr auto value = is_floating_point<_Tp>::value
+#ifdef __cuda_std__
+#ifndef _LIBCUDACXX_HAS_NO_NVFP16
+        || is_same<_Tp, __half>::value
+#ifndef _LIBCUDACXX_HAS_NO_NVBF16
+        || is_same<_Tp, __nv_bfloat16>::value
+#endif
+#endif
+#endif
+    ;
+};
```

Comment:
If we were to move the machinery out to respective header files we could change this to
```suggestion
template<class _Tp>
struct __is_nvfp16 : false_type {};
template<class _Tp>
struct __is_nvbf16 : false_type {};
template<class _Tp>
struct __is_complex_float : conjunction<is_floating_point<_Tp>, __is_nvfp16<_Tp>, __is_nvbf16<_Tp>> {};
```